### DISCUSSION TITLE
Why search latency has such difference on different querynodes?

### DISCUSSION THREAD

**Discussion by xiaojuan8** [milvus-io/milvus#22170] [Q&A and General discussion]
Hi, I found that search latency is not quite equal on querynodes. What is the key point leading to much diffs between searching latency on different nodes? 

<img width="1127" alt="image" src="https://user-images.githubusercontent.com/74544828/218672666-6511c88a-0c3e-4e6b-a759-656d09d84746.png">

**Comment by yhmo** [comment]
Load unbalance can lead to different latency on different query node.
Assume there are 3 segments in a collection, but only 2 query nodes in the cluster. Milvus will load segment_1 and segment_2 into node_1, and load segment_3 to node_2. Then node_1 will be slower than node_2.

**Reply by xiaojuan8** [reply]
But the latencies differs too much.  I have 16 querynodes, and they are evenly distributed on two server. They should be evenly loaded theoretically.
![image](https://user-images.githubusercontent.com/74544828/218693712-c148682c-39ef-4820-865b-f8af959688c1.png)

**Comment by yhmo** [comment]
How many collections? Search on one collection or search on multiple collections?
How many rows in each collection? What is the dimension value?

**Reply by xiaojuan8** [reply]
1. only 1 collection
2. Seach on 1 collection
3. 100million rows
4. dimension ： 32

**Reply by yhmo** [reply]
dim 32, 100m rows, the raw data size is 32*4*100m = 12.5GB
By default configuration, each segment size limit is 512MB, then there are about 25 segments in the collection.
There are 16 query nodes, some nodes might load 1 segment, some nodes might load 2 segments, workload might be uneven.

You can use the sizing tool to estimate querynode count: https://milvus.io/tools/sizing
I think your dataset only need 2 or 4 query nodes

**Reply by xiaojuan8** [reply]
OK, I try to resize replicas, thanks~
